README.md

```diff
@@ -1,139 +1,182 @@
-    case 'COUNTER_INCREMENTED':
+    case 'USER_REMOVED':
       return {
-        ...aggregate,
-        version,
-        count: aggregate.count + 1,
-      };
-    case 'COUNTER_REMOVED':
-      return {
-        ...aggregate,
+        ...userAggregate,
         version,
         status: 'REMOVED',
       };
   }
 };
+
 ```
+### Storage Adapter
 
-Finally, initialize an EventStore class and start interacting with your event store:
+!['toto'](/assets/storage_adapter_schema.png)
 
-```typescript
-const counterEventStore = new EventStore({
-  eventStoreId: 'Counters',
-  eventTypes,
-  reducer,
-  // 👇 See storage adapters section
-  storageAdapter,
-});
+You can store your events in many different ways. To specify how to store them (in memory, DynamoDB...) Castore implements Storage Adapters.
 
-const {
-  events, // <= Typed events
-  aggregate, // <= Reducer result
-} = await counterEventStore.getAggregate(aggregateId);
-
-// 👇 Method input is correctly typed
-await counterEventStore.pushEvent({
-  aggregateId: '123',
-  version: 1,
-  type: 'COUNTER_CREATED',
-  timestamp: new Date().toISOString(),
-  payload: {
-    startCount: 18,
-  },
-});
-```
+Adapters offer an interface between the Event Store class and your storage method 💾.
 
-# 🤔 Why use Castore ?
+To be able to use your EventStore, you will need to attach a Storage Adapter 🔗.
 
-- 💬 **Verbosity**: Castore classes are designed to increase dryness and provide the optimal developer experience. Event Sourcing is hard, don't make it harder!
-- 📝 **Strong typings**: We love type inference, we know you will to!
-- 🏄‍♂️ **Interfaces before implementations**: Castore provides a standard interface to modelize common event sourcing patterns in TypeScript. But it **DOES NOT enforce any particular implementation** (storage service, messaging system etc...). You can use Castore in React apps, containers or lambdas, it's up to you! Some common implementations are provided, but you are free to use **any implementation you want** via custom classes, as long as they follow the required interfaces.
-- 👍 **Enforces best practices**: Gained from years of usage like using int versions instead of timestamps, transactions for multi-store events and state-carrying transfer events for projections.
-- 🛠 **Rich suite of helpers**: Like mock events builder to help you write tests.
+All the Storage Adapters have the same interface, and you can create your own if you want to implement new storage methods!
 
-# Table of content
+So far, castore supports 2 Storage Adapters ✨:
+- in-memory
+- DynamoDB
 
-- [Event](#event)
-- [Event Store](#event-store)
-- [Storage Adapter](#storage-adapter)
-- [Mock Events Builder](#mock-events-builder)
+### Event Store Interface
+
+Now that our Event Store has been instantiated with a reducer and a Storage Adapter, we can start using it to actually populate our database with events and retrieve business data from it 🌈.
+
+To do that, the Event Store class exposes 2 methods:
```

Comment:
à ce stade on a l'impression que eventStore n'expose que 2 méthodes
exposes several methods, including the following two:
?